tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -380,25 +380,30 @@ tensorflow::Status CreateBroadcastableScalarConstant(
   return Status::OK();
 }
 
-// Convert an axis from TF format to TRT format while validating.
-tensorflow::Status ConvertAxis(int tf_axis, int trt_nb_dims, string node_name,
-                               int* trt_axis, bool allow_batch_dim = false) {
+// Convert an axis from TF format to TRT format while validating. TF format
+// includes the batch dimension, while TRT does not. TF can also use negative
+// indices.
+// TODO(tmorris): Use this method in more ops.
+tensorflow::Status ConvertAxis(int tf_axis, int trt_nb_dims, bool check_bounds,
+                               absl::string_view node_name, int* trt_axis) {
+  const int tf_nb_dims = trt_nb_dims + 1;
+  // Check bounds.
+  if (check_bounds && (tf_axis < -tf_nb_dims || tf_axis >= tf_nb_dims)) {
+    return tensorflow::errors::InvalidArgument(
+        "Axis value of ", std::to_string(tf_axis),
```

Comment:
we might not need to_string here as StrCat used in the error generation already handles the conversion.

Reply:
Oh you are correct

Reply:
we can drop to_string for tf_axis as well.